ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -153,31 +151,6 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
     onContentSizeChange();
   }
 
-  @Override
-  public boolean onTouchEvent(MotionEvent ev) {
-    switch (ev.getAction()) {
-      case MotionEvent.ACTION_DOWN:
-        mDetectScrollMovement = true;
-        // Disallow parent views to intercept touch events, until we can detect if we should be
-        // capturing these touches or not.
-        this.getParent().requestDisallowInterceptTouchEvent(true);
-        break;
-      case MotionEvent.ACTION_MOVE:
-        if (mDetectScrollMovement) {
-          if (!canScrollVertically(-1)
-              && !canScrollVertically(1)
-              && !canScrollHorizontally(-1)
```

Comment:
https://github.com/facebook/react-native/commit/372d001a5d5e24b68d194e01c356d973642fac86 does not detect direction of the scroll

>Note: this might change in the future to also detect the direction of the scroll, and
only block the scroll if the component can scroll in that direction. This is however not
trivial, since the scroll needs to be above some threshold of pixels. Blocking the
parent view from scrolling until that threshold is passed might cause incorrect behavior
in the parent component.



The checks [`canScroll`](https://developer.android.com/reference/android/view/View#canScrollVertically(int)) are performed for all directions (for ex. to scroll down, we check that TextInput can scroll Up, Left and Right), if the `TextInput` can not scroll in any direction, then the scroll `requestDisallowInterceptTouchEvent` is disabled and scrolling goes back to the parent. 

<img src="https://user-images.githubusercontent.com/24992535/83384610-e3a16700-a3e7-11ea-9e63-62ac4ed8c636.png" width="500" />

I would have to build a logic to detect the scroll direction and handle it correctly.

Reply:
The issue is caused by `canScrollHorizontally` returns allways `true` for:

```javascript
<TextInput multiline={false} textAlign="center" />
```

<img src="https://user-images.githubusercontent.com/24992535/83415729-db651e00-a41f-11ea-91a8-f5771964a5df.gif" /> 

The method definition of [`canScrollHorizontally`](https://github.com/aosp-mirror/platform_frameworks_base/blob/e80b45506501815061b079dcb10bf87443bd385d/core/java/com/android/internal/widget/ViewPager.java#L2380-L2394).

I'll prepare a workaround